src/KIM/kim_init.cpp

```diff
@@ -514,19 +523,25 @@ void KimInit::write_log_cite(char *model_name)
     return;
   }
 
+  auto pos = sm.prefix().length();
+  auto cite_openkim_potential = fmt::format("OpenKIM potential: "
+    "https://openkim.org/cite/{}#item-citation\n\n",
+    model_name_str.substr(pos, 19));
+
   for (int i = 0; i < extent; ++i) {
-    char const * fileName;
+    char const *fileName;
     int availableAsString;
-    char const * fileString;
+    char const *fileString;
     err = KIM_Collections_GetItemMetadataFile(
-        collections, i, &fileName, nullptr, nullptr,
-        &availableAsString, &fileString);
+      collections, i, &fileName, nullptr, nullptr,
+      &availableAsString, &fileString);
     if (err) continue;
 
-    if (0 == strncmp("kimcite", fileName, 7)) {
-      if ((lmp->citeme) && (availableAsString)) lmp->citeme->add(fileString);
-    }
+    if ((strncmp("kimcite", fileName, 7) == 0) and availableAsString)
+      cite_openkim_potential += fileString;
```

Comment:
@akohlmey is the concatenation safe here from your view or should I change it to `cite_openkim_potential += std::string(fileString);` or `cite_openkim_potential += fmt::format("{}", fileString);`

Reply:
yes, that is fine. I am more concerned about the use of `strncmp()`. That is error prone and difficult to debug. Better would be using `utils::strmatch(fileName,"^kimcite")` which will return true/false

Reply:
On second thought, there is a problem with this code. Not in how the string is constructed, but the fact that the string is going out of scope. The citeme class currently stores the address of the string pointer in a set to identify strings that have already been submitted. But that will break with this use. I will have to change that part of `CiteMe::add()` to use a hash instead. Then I can also make it use a `const std::string &` as argument and having temporaries added would not be a problem.

Reply:
PR #2629 should address my concerns about temporaries.